hazelcast-spring/src/main/java/com/hazelcast/spring/cache/HazelcastCacheManager.java

```diff
@@ -70,11 +83,59 @@ public Cache getCache(String name) {
         return cacheNames;
     }
 
-    public void setHazelcastInstance(final HazelcastInstance hazelcastInstance) {
-        this.hazelcastInstance = hazelcastInstance;
+    private long calculateCacheReadTimeout(String name) {
+        long timeout = 0;
+        if (getReadTimeoutMap().containsKey(name)) {
+            Long v = getReadTimeoutMap().get(name);
+            if (v != null) {
+                timeout = v;
+            }
+        } else {
+            timeout = defaultReadTimeout;
+        }
+        return timeout;
```

Comment:
Why don't we simply do as below? 
We should also guarantee that `readTimeoutMap` does not accept null values if a `null` received then we should fail fast on the startup...

if (getReadTimeoutMap().containsKey(name)) {
             return getReadTimeoutMap().get(name);
    } else {
             return defaultReadTimeout;
         }